### DISCUSSION TITLE
Select which URL's to open with OM

### DISCUSSION THREAD

**Discussion by bveldkamp** [organicmaps/organicmaps#4519] [Ideas]
Apparently there is a large number of URL's that open in Organic Maps. For instance, when clicking a link to maps.google.com in a browser, it opens in OM, rather than in the Google Maps app.

Sometimes this is fine, but at times I'd prefer using a different app. 

Is there a way to specify supported URL's, or can this be added to the app?

Btw, I'm on Android, and not sure if this is an issue on other platforms too.

Update 20 Feb 2023:
It seems that my issue is not with OM, but rather Firefox having its own list of default apps for certain links. Chrome on the other hand uses the system settings, which in my case is to open Google Maps.

**Comment by biodranik** [comment]
Can't you choose which url opens with which app? Also check this link: https://www.quora.com/How-can-I-open-links-on-a-respective-app-on-Android-instead-of-a-browser

**Comment by bveldkamp** [comment]
There's a setting in the android configuration screen (default apps), but that is all or nothing.

**Reply by biodranik** [reply]
Usually Android asks if a link should be opened once or always. If you select "once", then you can always choose which app to use for a link.

**Reply by bveldkamp** [reply]
Let's put this on hold for now, there are some settings that I need to investigate... And some links actually do open in Google Maps. Not sure what's going on

**Reply by bveldkamp** [reply]
Ok, so it appears that I *can* influence which app opens which link.  Here's what I notice: when I check the default links that OM opens, there's only one listed, omaps.app. There's an option to add more from a list of about 400. All are unchecked, and most of the Google links say 'opens in maps' or 'opens in Google'. This includes www.google.com

However, when I click the following link in Firefox, it still opens in OM: https://www.google.com/maps/d/u/1/edit?mid=17lbShssztRL48vxwo6mVRhQ2xLiwvMA&usp=sharing

Clicking the link in Chrome opens the Google Maps app. So it looks this is actually an error in the way Firefox handles links.

**Reply by rtsisyk** [reply]
Organic Maps doesn't open any external links by default since Android 11. This is a limitation of Android platform. You can enable or disable specific domains in Settings -> App -> Organic Maps -> Open by default -> Add Link (Android 11+):

<img src="https://user-images.githubusercontent.com/1799054/220054504-bfacfcd2-f5b2-4963-9d3d-5f4c9274452f.png" width="400px">

<img src="https://user-images.githubusercontent.com/1799054/220054526-cb049965-3785-4255-9718-f3650e6732e5.png" width="400px">

In the last years, Google update Google Maps apps to use goo.gl link shortener. Now, it is not even possible to open a link from Google Maps without querying Google's servers. Moreover, even resolved links has coordinates encoding via some proprietary algorithm. See https://github.com/organicmaps/organicmaps/issues/475

**Reply by bveldkamp** [reply]
Thanks all, it's clear to me now that this isn't an OM issue
